src/pytkdocs/loader.py

```diff
@@ -415,6 +431,49 @@ def get_property_documentation(self, node: ObjectNode) -> Attribute:
             source=source,
         )
 
+    def get_pydantic_field_documentation(self, node: ObjectNode) -> Attribute:
+        """
+        Get the documentation for a PyDantic Field
+
+        Arguments:
+            node: The node representing the Field and its parents.
+
+        Return:
+            The documented attribute object.
+        """
+        prop: ModelField = node.obj
+        path = node.dotted_path
+        properties = ["field"]
+        if prop.required:
+            properties.append("required")
+
+        return Attribute(
+            name=node.name,
+            path=path,
+            file_path=node.file_path,
+            docstring=prop.field_info.description,
+            attr_type=prop.type_,
+            properties=properties,
+        )
+
+    def get_annotated_dataclass_field(self, node: ObjectNode) -> Attribute:
+        """
+        Get the documentation for an dataclass annotation.
+
+        Arguments:
+            node: The node representing the annotation and its parents.
+
+        Return:
+            The documented attribute object.
+        """
+        annotation: type = node.obj
+        path = node.dotted_path
+        properties = ["field"]
+
+        return Attribute(
+            name=node.name, path=path, file_path=node.file_path, attr_type=annotation, properties=properties
```

Comment:
We'd need to parse a potential triple-quoted docstring below the declaration. But honestly this would need a refactor of the attribute parser module (I have improvements plan), so I'm willing to merge as-is and do the refactor myself :+1:

Reply:
I'm not even sure to go about this so, I think that might be best for now. It will work with dataclasses, it just won't catch the docstring.